### PR TITLE
Helping runtime diagnose a CI bug

### PR DESCRIPTION
**Pull request by jaredpar**

### CODE REVIEW

File: src/Tools/Source/RunTests/TestRunner.cs [hunk 1]
```diff
@@ -227,6 +227,13 @@ string MakeHelixWorkItemProject(WorkItemInfo workItemInfo)
                     command.AppendLine($"{setEnvironmentVariable} DOTNET_EnableCrashReport=1");
                 }
 
+                // https://github.com/dotnet/runtime/issues/97186#issuecomment-1899748532
+                // Helping to see if this is reproducible on Linux by disabling GC regionns
+                if (isUnix)
+                {
+                    command.AppendLine($"{setEnvironmentVariable} DOTNET_GCName=clrgc");
```
Comment (EgorBo) [inline src/Tools/Source/RunTests/TestRunner.cs:234]: I think this should be `libclrgc.so`? Not sure it will resolve as-is. cc @cshung

Comment (jaredpar) [reply src/Tools/Source/RunTests/TestRunner.cs:234]: If it doesn't resolve will the runtime error or fall back?

Comment (jaredpar) [reply src/Tools/Source/RunTests/TestRunner.cs:234]: Looks like it fails to start

```
/datadisks/disk1/work/9F910860/p/dotnet-cli/sdk/8.0.100/vstest.console.dll
+ dotnet exec /datadisks/disk1/work/9F910860/p/dotnet-cli/sdk/8.0.100/vstest.console.dll @vstest_1.rsp
GC initialization failed with error 0x8007007E
Failed to create CoreCLR, HRESULT: 0x8007007E
```

Comment (jaredpar) [reply src/Tools/Source/RunTests/TestRunner.cs]: Trying again with `libclrgc.so`

Comment (jaredpar) [reply src/Tools/Source/RunTests/TestRunner.cs]: Looks like Linux passes with that setting enabled.

File: src/Tools/Source/RunTests/TestRunner.cs [hunk 2]
```diff
@@ -225,6 +225,26 @@ string MakeHelixWorkItemProject(WorkItemInfo workItemInfo)
                     command.AppendLine($"{setEnvironmentVariable} DOTNET_DbgEnableMiniDump=1");
                     command.AppendLine($"{setEnvironmentVariable} DOTNET_DbgMiniDumpType=1");
                     command.AppendLine($"{setEnvironmentVariable} DOTNET_EnableCrashReport=1");
+
+                    // Find the eng directory
+                    var startPath = Path.GetDirectoryName(typeof(TestRunner).Assembly.Location)!;
+                    var path = Path.GetDirectoryName(startPath)!;
+                    var gcPath = Path.Combine(path, "eng", "libclrgc.dynlib");
```
Comment (cshung) [inline src/Tools/Source/RunTests/TestRunner.cs:232]: Should this be `dylib` instead?

Comment (jaredpar) [reply src/Tools/Source/RunTests/TestRunner.cs:232]: image

Comment (jaredpar) [reply src/Tools/Source/RunTests/TestRunner.cs]: Yeah it probably helps if I spell the file name correctly.

**Pull request closed without merging**